docs.go

```diff
@@ -0,0 +1,153 @@
+package cli
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/cpuguy83/go-md2man/md2man"
+)
+
+// ToMarkdown creates a markdown string for the `*App`
+// The function errors if either parsing or writing of the string fails.
+func (a *App) ToMarkdown() (string, error) {
+	var w bytes.Buffer
+	if err := a.writeDocTemplate(&w); err != nil {
+		return "", err
+	}
+	return w.String(), nil
+}
+
+// ToMan creates a man page string for the `*App`
+// The function errors if either parsing or writing of the string fails.
+func (a *App) ToMan() (string, error) {
+	var w bytes.Buffer
+	if err := a.writeDocTemplate(&w); err != nil {
+		return "", err
+	}
+	man := md2man.Render(w.Bytes())
+	return string(man), nil
+}
+
+type CliTemplate struct {
+	App          *App
+	Date         string
+	Commands     []string
+	GlobalArgs   []string
+	SynopsisArgs []string
+}
+
+func (a *App) writeDocTemplate(w io.Writer) error {
+	now := time.Now()
+	const name = "cli"
+	t, err := template.New(name).Parse(MarkdownDocTemplate)
+	if err != nil {
+		return err
+	}
+	return t.ExecuteTemplate(w, name, &CliTemplate{
+		App:          a,
+		Date:         fmt.Sprintf("%s %d", now.Month(), now.Year()),
+		Commands:     prepareCommands(a.Commands, 0),
+		GlobalArgs:   prepareArgsWithValues(a.Flags),
+		SynopsisArgs: prepareArgsSynopsis(a.Flags),
+	})
+}
+
+const nl = "\n"
+const noDescription = "_no description available_"
+
+func prepareCommands(commands []Command, level int) []string {
+	coms := []string{}
+	for i := range commands {
+		command := &commands[i]
+		prepared := strings.Repeat("#", level+2) + " " +
+			strings.Join(command.Names(), ", ") + nl
+
+		usage := noDescription
+		if command.Usage != "" {
+			usage = command.Usage
+		}
+		prepared += nl + usage + nl
```

Comment:
Use a string buildee for good practice, this is a ton of copying for no reason.

Reply:
Okay, used `fmt.Sprintf` and combined some statements together. :+1: